OSDP: Maintainers, samples and board updates

sidcha opened the pull request:
Ripping off individual commits that don't depend on anything from #46449.

sidcha committed:
samples: mgmt/osdp: Select ENTROPY_GENERATOR from prj.conf

ENTROPY_GENERATOR is needed for OSDP Secure Channel capability. Now
that Secure Channel patches are merged, enabling it by default.

Signed-off-by: Siddharth Chandrasekaran <sidcha.dev@gmail.com>

sidcha committed:
MAINTAINERS: Mark subsys/mgmt/osdp as maintained

subsys/mgmt/osdp _has_ a dedicated maintainer who will rush to address
any issues identified :)

Also, add @adakus and @r2r0 as collaborators since they are using this
subsystem in production and have agreed to review, test, and report issues
for this subsystem in future.

Signed-off-by: Siddharth Chandrasekaran <sidcha.dev@gmail.com>

erwango commented on boards/arm/nucleo_g474re/nucleo_g474re.dts:
```diff
@@ -19,6 +19,7 @@
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 		zephyr,canbus = &can1;
+		zephyr,osdp-uart = &usart1;
```
Since this chosen symbol is quite specific and only used in a single board, I wonder if this should rather be added as an overlay in the samples.
I think this would benefit to the sample to illustrate how to use it.
Could also be set as an `app.overlay` file and be more generic using arduino node label.
```
	chosen {
		zephyr,osdp-uart = &arduino_serial;
         };
```

sidcha replied:
@erwango, I like the idea of using `arduino_serial` node label. For boards that don't have it, is there a way to fallback to another?

For instance I use the [stm32_min_dev_(blue|black)](https://docs.zephyrproject.org/2.6.0/boards/arm/stm32_min_dev/doc/index.html) boards for testing but they don't have it.

erwango replied:
For boards that don't have it, you can add a board specific overlay to the sample.

MaureenHelm reviewed (commented):
Please address @erwango 's comments, but otherwise LGTM

MaureenHelm commented:
cc: @adakus @r2r0

sidcha committed:
mgmt/osdp: Add dt-chosen "osdp-uart" from app.overlay

Since the dt choosen property, "zephyr,osdp-uart" is specific to one this
subsystem, it makes sense to allow the samples to add it as an overlay.

Signed-off-by: Siddharth Chandrasekaran <sidcha.dev@gmail.com>

erwango reviewed (changes requested):
Please update `samples/subsys/mgmt/osdp/peripheral_device/sample.yaml` accordingly:
```
    depends_on: gpio arduino_serial
```

But this raises indeed and issue as in this case `stm32_min_dev_black` can't be used as integration platform.
Alternative is to drop the use of `arduino_serial`. This would limit the genericity of the sample, but adding a new overlay is not complex anyway.

erwango commented on samples/subsys/mgmt/osdp/control_panel/sample.yaml:
```diff
@@ -4,7 +4,7 @@ sample:
-    depends_on: gpio
+    depends_on: gpio arduino_serial
     filter: dt_enabled_alias_with_parent_compat("led0", "gpio-leds") and
             dt_chosen_enabled("zephyr,osdp-uart") and CONFIG_SERIAL
```
filter is applied before dtc compilation, hence before the overlay is applied.
`dt_chosen_enabled("zephyr,osdp-uart")` will always return false.

sidcha replied:
So how can I filter platforms that aren't supported? removing `dt_chosen_enabled("zephyr,osdp-uart")` widens the test space and so the test fails on unsupported platforms.

erwango replied:
Well, in theory "filter" should list allow to filter out al platforms that don"'t provide required dependencies, so in theory it shouldn't.
But I agree that on this area, practice often defeats the theory.  Out of curiosity, do you have an example of unsupported platform that is not filtered out ?

sidcha replied:
Just removing `dt_chosen_enabled("zephyr,osdp-uart")` from the filter causes test failure because the device tree lookup for "zephyr,osdp-uart".

I don't understand the failures as the app overlays should be applied at the time (after this filter is processed).

erwango replied:
Can you provide an example of failing test command so I can have a try?

sidcha replied:
Actually that is part of the problem; I cannot run the tests locally because something is broken in my python setup. Haven't spent the time fixing it.

Removing `dt_chosen_enabled("zephyr,osdp-uart")` caused CI builds to fail. I'll do that now.

sidcha replied:
@erwango, can you help me with this CI failure please :)

str4t0m commented on samples/subsys/mgmt/osdp/peripheral_device/sample.yaml:
```diff
@@ -4,7 +4,7 @@ sample:
   sample.mgmt.osdp.peripheral_device:
     tags: osdp
-    depends_on: gpio
+    depends_on: gpio arduino_serial
```
a

str4t0m replied:
same here.

str4t0m commented on samples/subsys/mgmt/osdp/control_panel/sample.yaml:
```diff
@@ -4,7 +4,7 @@ sample:
   sample.mgmt.osdp.control_panel:
     tags: osdp
-    depends_on: gpio
+    depends_on: gpio arduino_serial
```
This test now depends on `arduino_serial`, therefore can't use stm32_min_dev_black as an integration platform.
Please use a different board.

sidcha replied:
Switched to `nucleo_g474re` at both places. Thanks!

sidcha commented:
Maintainers and samples fix moved to separate PRs. The app overlay change needs a few more commits to clean up zephy-uart before it can be merged. So closing this PR.
